drivers/ethernet/eth_stm32_hal.c

```diff
@@ -38,6 +39,21 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include "eth.h"
 #include "eth_stm32_hal_priv.h"
 
+#if CONFIG_ETH_STM32_HAL_MAC3 || \
+	CONFIG_ETH_STM32_HAL_MAC4 || \
+	CONFIG_ETH_STM32_HAL_MAC5
+#warning CONFIG_ETH_STM32_HAL_MAC* is deprecated, use device tree "local-mac-address"
```

Comment:
This is not how deprecation is done - see https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated